solutions/banking/main.tf

```diff
@@ -116,8 +119,12 @@ module "configure_wml_project" {
   watson_ml_instance_resource_name = var.watson_machine_learning_instance_resource_name
   watson_ml_project_name           = local.watson_ml_project_name
   resource_group_id                = module.resource_group.resource_group_id
-  cos_instance_name                = "${var.prefix}-rag-sample-app-cos"
-  location                         = var.watson_discovery_region # WatsonX services needs to be in the same region anyway
+  cos_instance_name                = local.cos_instance_name
+  cos_kms_crn                      = var.cos_kms_crn
```

Comment:
I think you should also pass the watsonx_project_delegated parameter based on whether KMS CRN is provided or not